### ISSUE TITLE
powershell.exe started from pwsh.exe give preference to pwsh modules and fails module version compatiblity check

### ISSUE THREAD

**Issue by adityapatwardhan** [PowerShell/PowerShell#7324]
Windows PowerShell started from a pwsh.exe from preview.4 gives preference to pwsh module path.
As an effect, module load fails with `Modules_InsufficientPowerShellVersion,Microsoft.PowerShell.Commands.ImportModuleCommand`

Steps to reproduce
------------------

```powershell
PS D:\> powershell
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

PS D:\> Write-Verbose "Test" -verbose
Write-Verbose : The 'Write-Verbose' command was found in the module 'Microsoft.PowerShell.Utility', but the module could not be loaded. For more information, run 'Import-Module
Microsoft.PowerShell.Utility'.
At line:1 char:1
+ Write-Verbose "Test" -verbose
+ ~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Write-Verbose:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CouldNotAutoloadMatchingModule

PS D:\> Import-Module Microsoft.PowerShell.Utility
Import-Module : The version of Windows PowerShell on this computer is '5.1.18200.1000'. The module 'C:\program
files\powershell\6-preview\Modules\Microsoft.PowerShell.Utility\Microsoft.PowerShell.Utility.psd1' requires a minimum Windows PowerShell version of '6.1' to run. Verify that you have the minimum
required version of Windows PowerShell installed, and then try again.
At line:1 char:1
+ Import-Module Microsoft.PowerShell.Utility
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (C:\program file...ll.Utility.psd1:String) [Import-Module], InvalidOperationException
    + FullyQualifiedErrorId : Modules_InsufficientPowerShellVersion,Microsoft.PowerShell.Commands.ImportModuleCommand
```

**Workaround**
After starting Windows PowerShell adjust the PSModulePath

```PowerShell
$env:Psmodulepath = "C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules;$env:PSModulePath"
```

cc @rjmholt 

Expected behavior
-----------------

```none
Write-Verbose should work
```

Actual behavior
---------------

```none
Modules_InsufficientPowerShellVersion,Microsoft.PowerShell.Commands.ImportModuleCommand
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-preview.4
PSEdition                      Core
GitCommitId                    6.1.0-preview.4
OS                             Microsoft Windows 10.0.18200
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by rjmholt** [comment]
To document some offline discussions further:

This is a consequence of adding the Windows PowerShell System32 module directory to the pwsh module path:

* Start WindowsPowerShell
* It sees the System32 module path already on the PSModulePath
* The System32 module path is not prepended (even secretly) to the Windows PowerShell PSModulePath
* Modules attempt to be (auto)loaded from (e.g.) Microsoft.PowerShell.Utility
* The first module seen is a PowerShell Core 6 Microsoft.PowerShell.Utility
* The version check fails
* Loading fails and `gci` doesn't work

Possible long-term workarounds are:

1. Check native executions for `powershell.exe` and if so, remove System32 from the module path, or
2. Treat the System32 module path as a "hidden" PSModulePath component in PowerShell Core, so that it doesn't actually get added to the PSModulePath, but is just treated as such, so it isn't inherited.

**Comment by rjmholt** [comment]
Worth noting that (2) would also solve #7327.

Also, any solution to this would need to work with any flags `powershell.exe` might be started with, like `-noprofile`

**Comment by BradBarnich** [comment]
I hit this too. Was using pwsh as my main shell on windows, but some of our build scripts are hardcoded to powershell. 

With preview 4, they fail with messages like:
`Get-Date : The 'Get-Date' command was found in the module 'Microsoft.PowerShell.Utility', but the module could not be loaded. For more information, run 'Import-Module Microsoft.PowerShell.Utility'.`

**Comment by BradBarnich** [comment]
Another scenario, running VS code from pwsh ,`code .`, then opening the integrated terminal. Same issue

**Comment by daxian-dbw** [comment]
@rjmholt I'm afraid adding the `System32` module path to the end of `env:PSModulePath` is not the root cause. If your run `import-module Microsoft.PowerShell.Management`, you will see the following error:

```
PS C:\Users\dongbow\Downloads\PowerShell-6.1.0-preview.4-win-x64> import-module Microsoft.PowerShell.Management
import-module : The version of Windows PowerShell on this computer is '5.1.17134.165'. The module 'C:\users\dongbow\dow
nloads\powershell-6.1.0-preview.4-win-x64\Modules\Microsoft.PowerShell.Management\Microsoft.PowerShell.Management.psd1'
 requires a minimum Windows PowerShell version of '6.1' to run. Verify that you have the minimum required version of
Windows PowerShell installed, and then try again.
At line:1 char:1
+ import-module Microsoft.PowerShell.Management
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ResourceUnavailable: (C:\users\dongbo...Management.psd1:String) [Import-Module], Invalid
   OperationException
    + FullyQualifiedErrorId : Modules_InsufficientPowerShellVersion,Microsoft.PowerShell.Commands.ImportModuleCommand
```

The error indicate that the modules are failing because we changed `PowerShellVersion="3.0"` to `PowerShellVersion="6.1"`. I guess we need to change it back for all PSCore built-in modules.

**Comment by iSazonov** [comment]
I wonder why do we even look for standart/core modules by `env:PSModulePath`? If we hard-code the path to <pwsh.exe dir>\Modules internally and remove it from  `env:PSModulePath`  the problem will go away.

**Comment by SteveL-MSFT** [comment]
@iSazonov I think you should put that feedback in https://github.com/PowerShell/PowerShell-RFC/pull/133

**Comment by iSazonov** [comment]
@SteveL-MSFT Will do. 
Here my comment is for #7365

**Comment by iSazonov** [comment]
Fix in #7365 is temporary and it will be replaced later.
